### ISSUE TITLE
Partial credit for features available in strict mode 

### ISSUE THREAD

**Issue by paulirish** [compat-table/compat-table#575]
First, there's a distinction of implementation status worth clarifying:
- **Features behind a flag**. Developers cannot enable them for their users, but they can enable them for themselves. It's useful for the table to communicate so developers can experiment, but understand it's not production ready.
- **Features behind strict mode**. Developers _can_ enable these features for their users if they ship code in strict mode. This is a reasonable solution for many sites and apps. Moreover it's very reasonable for javascript in node modules or browser extensions.

I hesitate to estimate how much of the audience of the compat-table ships in strict mode regularly, but I'd wager it's a worthwhile percentage.

<hr>

Picking up on the threads in #574 and #498, I think the above now clarifies that a strict-only feature opens its production availability to some definite audience segment. 

Given that, is it worth providing partial credit (~20%) to features available in strict but not yet in sloppy? 

_(I wouldn't expect <img src="https://cloud.githubusercontent.com/assets/39191/8487930/3b0bd6b2-20c4-11e5-83df-f42aaa4d7a09.png" height=18> to change its appearance or score, but rather the percentage at the top. )_

**Comment by kangax** [comment]
I'm ok with giving "implemented under strict" partial credit (like 20%); let's do it.

/cc @webbedspace @ljharb @zloirock

**Comment by ljharb** [comment]
Sure, that seems reasonable.

**Comment by webbedspace** [comment]
Hm, I'm of two minds on this. You may notice that one set of tests - `const` and `let` - currently do give 50% "partial credit" for strict mode, because they have explicit split tests for strict and non-strict. (This is because the actual results for Chrome and Firefox require the intersection of multiple factors: Chrome requires strict mode + the experimental flag, Firefox (at least for "for-loop iteration scope") requires strict mode + the `version=1.7` attribute.) Ideally, every other test with strict results should award the same 50% partial credit, too.

However, I feel like giving strict mode special consideration over other kinds of "flagged" results may open the door for giving partial credit to other kinds of bizarre non-conforming behaviour - for instance, some Firefox Nightly `class` implementations have only worked if the `constructor` is always user-provided - the burden of the tests checking for which would be increasingly wearying and would basically amount to encoding transient implementation quirks into the tests themselves. So, that has until now given me pause.

**Comment by zloirock** [comment]
Makes sense, but only strict mode, not "other kinds of bizarre non-conforming behaviour" :)

**Comment by Ajedi32** [comment]
Yeah, while I understand "incorrectly requires strict mode" as being a problem in that the feature isn't strictly standards compliant, from a practical perspective I'm not sure how big of a difference it actually makes. If I'm writing code that uses new ES6 features, chances are that ensuring said code is compliant with strict mode isn't going to be a big hassle for me anyway.

Also (and this is just my personal opinion) I actually kind of like the restriction, since it encourages adoption of strict mode in modern codebases. The way I see it, strict mode was designed to fix a bunch of outstanding problems with JavaScript in a way that doesn't break backwards compatibility for existing scripts. It's the web equivalent of a release with breaking changes in other languages, and, by extension, "loose mode" is the equivalent of a legacy version of such a language.

If we could start moving towards a future where it's viable for browsers to start enforcing "strict mode" by default (though obviously that's a long ways away), that'd be a win as far as I'm concerned.
